substrate/client/consensus/aura/src/import_queue.rs

```diff
@@ -287,129 +380,126 @@ where
 	}
 }
 
-/// Should we check for equivocation of a block author?
-#[derive(Debug, Clone, Copy)]
-pub enum CheckForEquivocation {
-	/// Yes, check for equivocation.
-	///
-	/// This is the default setting for this.
-	Yes,
-	/// No, don't check for equivocation.
-	No,
-}
-
-impl CheckForEquivocation {
-	/// Should we check for equivocation?
-	fn check_for_equivocation(self) -> bool {
-		matches!(self, Self::Yes)
-	}
-}
-
-impl Default for CheckForEquivocation {
-	fn default() -> Self {
-		Self::Yes
-	}
-}
-
 /// Parameters of [`import_queue`].
-pub struct ImportQueueParams<'a, Block: BlockT, I, C, S, CIDP> {
+pub struct ImportQueueParams<'a, B: BlockT, I, C, SC, S, CIDP> {
 	/// The block import to use.
 	pub block_import: I,
 	/// The justification import.
-	pub justification_import: Option<BoxJustificationImport<Block>>,
+	pub justification_import: Option<BoxJustificationImport<B>>,
 	/// The client to interact with the chain.
 	pub client: Arc<C>,
+	/// Chain selection system.
+	pub select_chain: SC,
 	/// Something that can create the inherent data providers.
 	pub create_inherent_data_providers: CIDP,
 	/// The spawner to spawn background tasks.
 	pub spawner: &'a S,
 	/// The prometheus registry.
 	pub registry: Option<&'a Registry>,
 	/// Should we check for equivocation?
-	pub check_for_equivocation: CheckForEquivocation,
+	pub check_for_equivocation: bool,
```

Comment:
This is redundant.  check_for_equivocation = offchain_tx_pool_factory.is_some()?